Trend Micro/trend-micro-apex-one/ingest/parser.yml

```
@@ -0,0 +1,618 @@
+    filter: "{{parsed_event.message.Name == 'Suspicious Connection'}}"
+  - name: set_pattern_update_status_event
+    filter: "{{parsed_event.message.Name == 'Pattern Update Status'}}"
+  - name: set_predictive_machine_learning_event
+    filter: "{{parsed_event.message.Name == 'virusa'}}"
+  - name: set_sandbox_detection_event
+    filter: "{{parsed_event.message.Name == 'VAN_RANSOMWARE.umxxhelloransom_abc'}}"
+  - name: set_spyware_grayware_event
+    filter: "{{parsed_event.message.Name == 'Spyware Detected'}}"
+  - name: set_suspicious_files_event
+    filter: "{{parsed_event.message.Name == 'Suspicious Files'}}"
+  - name: set_virus_malware_event
+    filter: "{{parsed_event.message.Name == 'JS_EXPLOIT.SMDN' or parsed_event.message.Name == 'WORM_GAMARUE.ITK'}}"
+  - name: set_web_security_event
+    filter: "{{parsed_event.message.Name == '7'}}"
+
+stages:
+  set_timestamp:
+    actions:
+      # Default timestamp is the start or startTime field
+      - set:
+          "@timestamp": "{{parse_start_date.date}}"
+
+      # Second choice is the receipt time (rc) field
+      - set:
+          "@timestamp": "{{parse_receipt_time.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time != null}}"
+
+      # Last choice is the end or endTime field
+      - set:
+          "@timestamp": "{{parse_end_date.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time == null and parse_end_date != null}}"
+
+  set_cef_header_fields:
+    # CEF:Version|Device Vendor|Device Product|Device Version|Device Event Class ID|Name|Severity|[Extension]
+    actions:
+      - set:
+          observer.vendor: "{{parsed_event.message.DeviceVendor}}"
+          observer.product: "{{parsed_event.message.DeviceProduct}}"
+          event.dataset: "{{parsed_event.message.Name}}"
+          observer.version: "{{parsed_event.message.DeviceVersion}}"
+          rule.id: "{{parsed_event.message.DeviceEventClassID}}"
+
+      - set:
+          # Match only for agentSeverity of type int
+          event.severity: "{{parsed_event.message.Severity}}"
+        filter: "{{parsed_event.message.Severity | re_match('[1-9]|10')}}"
+
+  set_attack_discovery_detection_event:
+    actions:
+      - set:
+          event.kind: alert
+          event.category: ["intrusion_detection"]
+          event.type: ["info"]
+          user.name: "{{parsed_event.message.duser}}"
+          source.ip: "{{parsed_event.message.c6a3}}"
+          destination.ip: "{{parsed_event.message.dst}}"
+          host.name: "{{parsed_event.message.dhost}}"
+          rule.name: "{{parsed_event.message.SLF_RuleID}}"
+          rule.ruleset: "{{parsed_event.message.cat}}"
+
+  set_behavior_monitoring_event:
+    actions:
+      - set:
+          event.kind: event
+          event.category: ["process"]
+          host.name: "{{parsed_event.message.dvchost}}"
+          source.ip: "{{parsed_event.message.src}}"
+          process.executable: "{{parsed_event.message.sproc}}"
```

I suggest adding a custom field (must be declared in the taxonomy) to extract the target of the behavior:
```suggestion
          process.executable: "{{parsed_event.message.sproc}}"
          trendmicro.apexone.behavior.target: "{{parsed_event.message.Target}}"
```